### PR TITLE
Add option to pass additional/custom information to notification scripts

### PR DESCRIPTION
**Pull request by martinwe-adfinis**
##### SUMMARY

In our environment, notification SMSes are sent to different users depending on the team to which a service is assigned *and* the team in which a user is.

We also send a copy of the SMS to other communication channels (e.g. Mattermost) for easier tracking/interaction, but in that case, all notifications (regardless of the assigned team) end up in the same place.

It would be nice to have a way to filter notifications by teams (or based on any other custom information). These changes implement a way to do so.

##### ISSUE TYPE

 - Feature Pull Request

##### ANSIBLE VERSION

```
ansible [core 2.12.10]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/martinwe/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /home/martinwe/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110]
  jinja version = 2.11.3
  libyaml = True
```

##### ADDITIONAL INFORMATION

There is now an additional commandline option (`-X`), which is optional. It takes any string as argument.

The script continues to work if the command template, command, notification template and notifications in Icinga2 have not been set up to use the option. In fact, the script by itself does not do anything with the additional option&mdash;it is merely a way to communicate additional information to the playbook author (for use in `icinga2_master_twilio_sms_additional_commands`).

To make use of the option, it is recommended to check if the option has been passed with a non-empty value; most easily something like:

```yaml
icinga2_master_twilio_sms_additional_commands:
  - …
  - 'if [ -n "$EXTRA_DATA" ]; then …; fi'
  - …
```

### CODE REVIEW

File: templates/etc/icinga2/scripts/twilio-sms-service-notification.sh.j2
Comment (martinwe-adfinis) [inline templates/etc/icinga2/scripts/twilio-sms-service-notification.sh.j2:33]: The `SLA_` prefix might be pointless. Thinking about removing it.

In fact, we can maybe even generalise this option to just take "any further information", to be used in the user-provided `icinga2_master_twilio_sms_additional_commands`.

**Review by martinwe-adfinis — COMMENTED** [review]
This only addresses services, but we can also get notifications for hosts. Would probably be worth adapting the host notification script in a similar fashion.

File: templates/etc/icinga2/scripts/twilio-sms-service-notification.sh.j2
Comment (martinwe-adfinis) [reply templates/etc/icinga2/scripts/twilio-sms-service-notification.sh.j2:33]: `-X` for "extra information" comes to mind. Any better ideas/suggestions? :slightly_smiling_face:

**Comment by gianklug** [comment]
Can you fix the Actions pipeline while you're already at the code? 🙂

Looks like the default branch has been renamed from `mastee` to `main`.

File: templates/etc/icinga2/scripts/twilio-sms-service-notification.sh.j2
Comment (martinwe-adfinis) [reply templates/etc/icinga2/scripts/twilio-sms-service-notification.sh.j2]: Now reading `-X` instead of `-S`, and renamed the corresponding variable to `EXTRA_DATA`.

**Comment by martinwe-adfinis** [comment]
Currently blocked on fulfilling the dependency `ansible-sygroup.icinga2_agent`. It looks like the pipeline can successfully clone the repo (though unsure if that's even necessary), but then `ansible-lint` fails to find the role in some step.

Do we have a working example of a role being tested in the GitHub CI? I'd love to compare what's different there; my (rather limited) GitHub CI foo is failing me.

(there's also the issue of a missing Zypper module, but that's for a second step)

**Comment by martinwe-adfinis** [comment]
Expanded the scope to also host notifications (it's weird to only be able to split notifications for services but not hosts).

**Comment by martinwe-adfinis** [comment]
@gianklug I have removed my unsuccessful attempts at fixing the pipeline; I believe that should be addressed in a separate PR and by people who understand the setup.

Let me know if this can be merged, or if you have any plans for how to proceed here.

(also, @madpat1998: ping, for input)

**Review by gianklug — APPROVED** [review]

**Pull request merged**
